src/Drupal/DrupalExtension/Context/DrupalContext.php

```diff
@@ -183,32 +183,134 @@ public function getTableRow(Element $element, $search)
+    }
+    $rows = array_filter($rows, static function ($row) use ($search) {
+      if (strpos($row->getText(), $search) !== FALSE) {
+        return $row;
+      }
+    });
+    if (empty($rows)) {
+      throw new \RuntimeException(sprintf('Failed to find a row containing "%s" on the page %s', $search, $this->getSession()->getCurrentUrl()));
+    }
+    return $rows;
+  }
+
+  /**
+   * See if the element has a table.
+   *
+   * @param Element $element
+   *    \Behat\Mink\Element\Element object.
+   *
+   * @return bool
+   *   True if the element has a table.
+   */
+  private function hasTable(Element $element): bool {
+    $rows = $element->findAll('css', 'tr');
+    return !empty($rows);
+  }
+
   /**
    * Find text in a table row containing given text.
    *
    * @Then I should see (the text ):text in the :rowText row
    */
-    public function assertTextInTableRow($text, $rowText)
-    {
-        $row = $this->getTableRow($this->getSession()->getPage(), $rowText);
-        if (strpos($row->getText(), $text) === false) {
-            throw new \Exception(sprintf('Found a row containing "%s", but it did not contain the text "%s".', $rowText, $text));
+    public function assertTextInTableRow($text, $rowText): void {
+      $rows = $this->getTableRows($this->getSession()->getPage(), $rowText);
+      foreach ($rows ?: [] as $row) {
+        if (strpos($row->getText(), $text) !== FALSE) {
+          return;
         }
+      }
+      throw new \RuntimeException(sprintf('Found a row containing "%s", but it did not contain the text "%s".', $rowText, $text));
+    }
+
+  /**
+   * Find text in all table rows containing given text.
+   *
+   * @Then I should see (the text ):text in all the :rowText rows
+   */
+  public function assertTextInTableRows($text, $rowText): void {
+    $rows = $this->getTableRows($this->getSession()->getPage(), $rowText);
+    foreach ($rows ?: [] as $row) {
+      if (strpos($row->getText(), $text) !== FALSE) {
+        throw new \RuntimeException(sprintf('Found a row containing "%s", but it did not contain the text "%s".', $rowText, $text));
+      }
     }
+  }
 
   /**
-   * Asset text not in a table row containing given text.
+   * Assert text not in a table row containing given text.
+   * If multiple rows are found, check the first one only.
    *
    * @Then I should not see (the text ):text in the :rowText row
    */
-    public function assertTextNotInTableRow($text, $rowText)
-    {
-        $row = $this->getTableRow($this->getSession()->getPage(), $rowText);
-        if (strpos($row->getText(), $text) !== false) {
-            throw new \Exception(sprintf('Found a row containing "%s", but it contained the text "%s".', $rowText, $text));
+  public function assertTextNotInTableRow($text, $rowText): void {
+    $row = $this->getTableRow($this->getSession()->getPage(), $rowText);
+    if (strpos($row->getText(), $text) !== false) {
```

Comment:
@kumikoono Please use `FALSE`.

```suggestion
    if (strpos($row->getText(), $text) !== FALSE) {
```